File: dotnet/Trinsic/Storage/ITokenProvider.cs
using System.Diagnostics.CodeAnalysis;
using System.Threading;
using System.Threading.Tasks;

namespace Trinsic;

[SuppressMessage("ReSharper", "ClassNeverInstantiated.Global")]
internal class TokenDefaults
{
    internal const string Name = "dotnet.authtoken";
}
public interface ITokenProvider
{
    Task<string?> GetAsync(string name = TokenDefaults.Name, CancellationToken cancellationToken = default);

    string? Get(string name = TokenDefaults.Name);

    Task SaveAsync(string authToken, string name = TokenDefaults.Name, CancellationToken cancellationToken = default);

    void Save(string authToken, string name = TokenDefaults.Name);
}

File: java/src/main/java/trinsic/storage/ITokenProvider.java
package trinsic.storage;

import com.google.common.util.concurrent.ListenableFuture;

public interface ITokenProvider {
  ListenableFuture<String> get(String name);

  ListenableFuture<String> save(String accountProfile, String name);

  ListenableFuture<String> save(String accountProfile);
}
